### PR TITLE
added activities to vessel profile

### PR DESCRIPTION
**Pull request by fertandil87**

### CODE REVIEW

File: apps/fishing-map/features/vessel/activity/activity-by-type/activity-by-type.hook.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    Math.ceil(events.filter((event) => !event?.subEvent).length ?? 0),
+  [EventTypes.Gap]: (events: RenderedEvent[]) => events.length ?? 0,
+}
+
+export const useActivityByType = () => {
+  //const eventsLoading = useSelector(selectEventResourcesLoading)
+  const eventsList = useSelector(selectFilteredEventsWithMainPortVisit)
+  const [expandedGroups, setExpandedGroups] = useState<EventType[]>([])
+  console.log(eventsList)
+  const eventsByType = useMemo(
```
Comment (j8seangel) [inline apps/fishing-map/features/vessel/activity/activity-by-type/activity-by-type.hook.ts:24]: I want all the grouping and filtering logic in the selector and not spread in the app. Also, this is not performant at all.
Could you take a look?

File: apps/fishing-map/features/vessel/activity/activity-by-type/activity-by-type.selectors.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      previous.push({
+        ...portVisitEvent,
+        description: `${t('event.port', 'Port')} ${
+          portLabel ?? t('event.unknown', 'Unknown event')
+        }`,
+        durationDescription:
+          portEntry?.durationDescription ?? portExit?.durationDescription ?? duration
+            ? [
+                duration.hours && duration.hours > 0
+                  ? t('event.hourAbbreviated', '{{count}}h', { count: duration.hours })
```
Comment (j8seangel) [inline apps/fishing-map/features/vessel/activity/activity-by-type/activity-by-type.selectors.ts:66]: Translations can't be used in selectors as changing the language doesn't trigger the update

**Review by j8seangel — CHANGES_REQUESTED** [review]
Some high-level comments, lets sync again once you have more time to polish and review

File: apps/fishing-map/features/vessel/VesselActivity.tsx [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
-        <Button className={styles.actionButton}>
-          {t('vessel.activityGroupByType', 'Group by type')}
+        <label>
+          {activityMode === 'voyages'
+            ? t('vessel.activityByVoyages', 'Timeline by voyages')
+            : t('vessel.activityGroupByType', 'Group by type')}
+        </label>
+        <Button
+          className={styles.actionButton}
+          onClick={(e) =>
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map/features/vessel/VesselActivity.tsx:30]: yes, no problem

File: apps/fishing-map/features/regions/regions.slice.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+const sortRegionAlphabetically = (a: Region, b: Region) => (a.label < b.label ? -1 : 1)
+
+export const fetchRegionsThunk = createAsyncThunk(
+  'regions/fetch',
+  async (_, { rejectWithValue }) => {
+    try {
+      const apiUrl = `/datasets`
+      const options = {}
+      const promises = [
+        GFWAPI.fetch<Region[]>(`${apiUrl}/public-eez-areas/user-context-layer-v1`, options),
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map/features/regions/regions.slice.ts:44]: I forget your idea about this, we can do it in a different PR

File: apps/fishing-map/features/vessel/activity/ActivityDataAndTerminology.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { Fragment } from 'react'
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map/features/vessel/activity/ActivityDataAndTerminology.tsx:1]: maybe yes, maybe no, I removed for the moment

File: apps/fishing-map/package.json [hunk 6]
```diff
@@ -61,6 +61,8 @@
     "react-ga": "3.3.1",
     "react-i18next": "12.3.0 ",
     "react-sticky-el": "2.0.9",
+    "react-virtualized-auto-sizer": "^1.0.7",
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map/package.json:64]: finally used for activity by type, not needed in activity by voyage

**Review by j8seangel — CHANGES_REQUESTED** [review]
In general, it seems a bit better, but I think there is still a lot of work of cleaning.
It is great to recover the virtualized list, but remember the event info is not in a modal.

File: apps/fishing-map/features/vessel/activity/activity-by-type/ActivityByType.tsx [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            events[eventType]?.events.map((event) => (
+              <ActivityItem
+                key={event.id}
+                event={event}
+                onMapClick={selectEventOnMap}
+                onInfoClick={openModal}
+                options={displayOptions}
+              />
+            ))}
+          {expandedGroup === eventType && events[eventType]?.events.length && (
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map/features/vessel/activity/activity-by-type/ActivityByType.tsx:123]: I didn't see any 0 but I take a look closer

File: apps/fishing-map/features/vessel/activity/activity-by-type/ActivityItem.tsx
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map/features/vessel/activity/activity-by-type/ActivityItem.tsx:1]: I removed logic so it wasn't doing nothing specially now

File: apps/fishing-map/features/vessel/activity/activity-by-voyage/activity-by-voyage.slice.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import { RootState } from 'store'
+import { RenderedVoyage } from 'types/voyage'
+
+export interface VoyagesState {
+  voyages: Record<number, RenderedVoyage>
+  initialized: boolean
+}
+const initialState: VoyagesState = {
+  voyages: {},
+  initialized: false,
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map/features/vessel/activity/activity-by-voyage/activity-by-voyage.slice.ts:11]: refactored all the slice

File: apps/fishing-map/features/vessel/activity/activity-by-voyage/activity-by-voyage.slice.ts
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map/features/vessel/activity/activity-by-voyage/activity-by-voyage.slice.ts:1]: yes, everything is different, the only common thing that they have is the final Event itself

**Pull request merged**